tests/nn/data_parallel/test_fsdp_metadata.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+import pytest
+import torch
+import torch.nn as nn
+
+from fairscale.nn import FullyShardedDataParallel
+from fairscale.utils.testing import in_temporary_directory, skip_if_single_gpu, temp_files_ctx
+
+
+class SimpleNestedModel(nn.Module):
+    def __init__(self, embedding_size: int, with_fsdp: bool, process_group):
+        super().__init__()
+        fc1 = nn.Linear(embedding_size, 2 * embedding_size)
+        fc2 = nn.Linear(2 * embedding_size, 2 * embedding_size)
+        fc3 = nn.Linear(2 * embedding_size, embedding_size + 1)
+        fc4 = nn.Linear(embedding_size + 1, embedding_size)
+        if with_fsdp:
+            self.fc1 = FullyShardedDataParallel(fc1, process_group=process_group)
+            self.fc2 = fc2  # To test different levels of nesting
+            self.fc3 = FullyShardedDataParallel(fc3, process_group=process_group, flatten_parameters=False)
+            self.fc4 = fc4
+        else:
+            self.fc1 = fc1
+            self.fc2 = fc2
+            self.fc3 = fc3
+            self.fc4 = fc4
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.fc2(x)
+        x = self.fc3(x)
+        x = self.fc4(x)
+        return x
+
+
+def _worker(gpu_id: int, sync_file: str, world_size: int, embedding_size: int, flatten_parameters: bool):
```

Comment:
```suggestion
def _worker(gpu_id, sync_file, world_size, embedding_size, flatten_parameters):
```

we usually don't type checking the test to lower a bit of burden on the devs.